src/Type/Php/InArrayFunctionTypeSpecifyingExtension.php

```diff
@@ -41,22 +43,36 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 			return new SpecifiedTypes([], []);
 		}
 
-		$arrayValueType = $scope->getType($node->getArgs()[1]->value)->getIterableValueType();
+		$arrayType = $scope->getType($node->getArgs()[1]->value);
+		$arrayValueType = $arrayType->getIterableValueType();
+
+		$specifiedTypes = new SpecifiedTypes([], []);
+		if ($context->true()) {
+			$arrayType = TypeCombinator::intersect($arrayType, new NonEmptyArrayType());
```

Comment:
it seems this interesection alters some array-types, so they no longer get reported as TRUE by the ImpossibleTypeChecker.. still investigating

Reply:
had to add a lot of checks, to make the test-suite - especially the ImpossibleTypeChecker - happy.

I am pretty sure, there is some kind of check which should be done instead, but I did not find it yet

Reply:
hmm maybe there is some kind of cache invalidated, because the type is altered within the type-specifying-extension?

e.g. intersecting `array{'foo', 'bar'}` with `NonEmptyArrayType` does not change the type itself, but it feels like since we get a new object this has some kind of side-effect on the ImpossibleTypeChecker..

Reply:
investigation still goes on... 

before this PR, phpstan took this path to `return false` 

https://github.com/phpstan/phpstan-src/blob/a355aaa147429b6ba07ab1c8ebc418b8b647127e/src/Rules/Comparison/ImpossibleCheckTypeHelper.php#L201-L220

which produced the errors, now missing with this PR applied. after this change we have more then 1 sureType, and therefore the code takes a different path.

Reply:
I am getting the feeling, that a `TypeSpecifyingExtension` cannot specify types for different expressions in one run.

I cannot find any other `TypeSpecifyingExtension` wich uses `unionWith`. also every other `TypeSpecifyingExtension` which I checked in the codebase always specifies its type for a single arg.